src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -1116,7 +1116,9 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
         case tpnme.ScalaATTR =>
           isScalaRaw = true
           i = numAttrs
-        case tpnme.TASTYATTR if YtastyReader =>
+        case tpnme.TASTYATTR if !YtastyReader =>
+          reporter.warning(NoPosition, s"Add -Ytasty-reader to scalac options to parse the TASTy in $file")
```

Comment:
Should it be an error? Treating Scala 3 classfiles like Java classfiles is not going to do any good.